src/main/java/com/example/git_repo/error/decoder/GitErrorDecoder.java

```diff
@@ -0,0 +1,18 @@
+package com.example.git_repo.error.decoder;
+
+import feign.Response;
+import feign.codec.ErrorDecoder;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+public class GitErrorDecoder implements ErrorDecoder {
+
+    @Override
+    public Exception decode(String methodKey, Response response) {
+        if (response.status() == HttpStatus.SERVICE_UNAVAILABLE.value()) {
```

Comment:
Odpowiedz mi kiedy robi sie retry dla klienta feignowego? I zastosuj zeby jednak kiedys sie ten retry robil bo inaczej to po co on nam?